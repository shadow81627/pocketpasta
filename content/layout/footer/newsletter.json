{
  "icon": "jam:newsletter",
  "name": "Newsletter",
  "route": "newsletter",
  "pos": "6/1"
}
