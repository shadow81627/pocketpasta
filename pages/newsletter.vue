<template>
  <div class="container">
    <div class="row">
      <div class="col">
        <h1>{{ $t('newsletter.heading') }}</h1>
      </div>
    </div>
    <!-- Begin MailChimp Signup Form -->
    <div id="mc_embed_signup">
      <form
        id="mc-embedded-subscribe-form"
        action="https://pocketpasta.us16.list-manage.com/subscribe/post?u=7f69c361c7f5d97725d2e339b&amp;id=c821016b08"
        method="post"
        name="mc-embedded-subscribe-form"
        class="validate"
        target="_blank"
        novalidate
      >
        <div id="mc_embed_signup_scroll">
          <div class="indicates-required text-right">
            <span class="asterisk text-danger">*</span>
            {{ $t('newsletter.required') }}
          </div>
          <div class="mc-field-group form-group pb-4">
            <label for="mce-EMAIL">
              {{ $t('newsletter.email.label') }}
              <span class="asterisk text-danger">*</span>
            </label>
            <input
              id="mce-EMAIL"
              type="email"
              name="EMAIL"
              class="form-control required email"
              :placeholder="$t('newsletter.email.placeholder')"
            />
          </div>
          <div class="mc-field-group form-group pb-4">
            <label for="mce-FNAME">{{
              $t('newsletter.firstname.label')
            }}</label>
            <input
              id="mce-FNAME"
              type="text"
              name="FNAME"
              class="form-control"
              :placeholder="$t('newsletter.firstname.placeholder')"
            />
          </div>
          <div class="mc-field-group form-group pb-4">
            <label for="mce-LNAME">{{ $t('newsletter.lastname.label') }}</label>
            <input
              id="mce-LNAME"
              type="text"
              name="LNAME"
              class="form-control"
              :placeholder="$t('newsletter.lastname.placeholder')"
            />
          </div>
          <div id="mce-responses" class="clear">
            <div
              id="mce-error-response"
              class="response"
              style="display:none"
            />
            <div
              id="mce-success-response"
              class="response"
              style="display:none"
            />
          </div>
          <!-- real people should not fill this in and expect good things
        - do not remove this or risk form bot signups-->
          <div style="position: absolute; left: -5000px;" aria-hidden="true">
            <input
              type="text"
              name="b_7f69c361c7f5d97725d2e339b_c821016b08"
              tabindex="-1"
              value
            />
          </div>
          <div class="clear">
            <input
              id="mc-embedded-subscribe"
              type="submit"
              :value="$t('newsletter.submit')"
              name="subscribe"
              class="button btn btn-primary"
            />
          </div>
        </div>
      </form>
    </div>
    <!--End mc_embed_signup-->
  </div>
</template>

<script>
export default {
  head() {
    return {
      link: [
        {
          hid: 'canonical',
          rel: 'canonical',
          href: 'https://pocketpasta.com/newsletter',
        },
      ],
    };
  },
  methods: {
    subscribe() {
      this.$ga.event('Newsletter', 'Click', 'Subscribe');
    },
  },
};
</script>

<style>
/* @import url('https://cdn-images.mailchimp.com/embedcode/classic-10_7.css'); */
</style>
