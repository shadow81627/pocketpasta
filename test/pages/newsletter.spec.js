import newsletter from '@/pages/newsletter.vue';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import Vuetify from 'vuetify';
import Vue from 'vue';

const localVue = createLocalVue();

Vue.use(Vuetify);

let vuetify;

const factory = () =>
  shallowMount(newsletter, {
    localVue,
    vuetify,
    mocks: {
      $ga: { event: () => true },
    },
  });

describe('newsletter', () => {
  beforeEach(() => {
    vuetify = new Vuetify();
  });

  test('mounts properly', () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test('renders properly', () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

  test('head', () => {
    expect(newsletter.head()).toBeTruthy();
  });

  test('subscribe event', () => {
    const wrapper = factory();
    expect(wrapper.vm.subscribe()).toBe();
  });
});
