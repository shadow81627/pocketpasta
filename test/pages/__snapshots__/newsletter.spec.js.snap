// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`newsletter renders properly 1`] = `
"<div class=\\"container\\">
  <div class=\\"row\\">
    <div class=\\"col\\">
      <h1>newsletter.heading</h1>
    </div>
  </div>
  <div id=\\"mc_embed_signup\\">
    <form id=\\"mc-embedded-subscribe-form\\" action=\\"https://pocketpasta.us16.list-manage.com/subscribe/post?u=7f69c361c7f5d97725d2e339b&amp;id=c821016b08\\" method=\\"post\\" name=\\"mc-embedded-subscribe-form\\" target=\\"_blank\\" novalidate=\\"novalidate\\" class=\\"validate\\">
      <div id=\\"mc_embed_signup_scroll\\">
        <div class=\\"indicates-required text-right\\"><span class=\\"asterisk text-danger\\">*</span>
          newsletter.required
        </div>
        <div class=\\"mc-field-group form-group pb-4\\"><label for=\\"mce-EMAIL\\">
            newsletter.email.label
            <span class=\\"asterisk text-danger\\">*</span></label>
          <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" id=\\"mce-EMAIL\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" placeholder=\\"newsletter.email.placeholder\\" type=\\"email\\" name=\\"EMAIL\\" class=\\"form-control required email\\"></v-text-field-stub>
        </div>
        <div class=\\"mc-field-group form-group pb-4\\"><label for=\\"mce-FNAME\\">newsletter.firstname.label</label>
          <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" id=\\"mce-FNAME\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" placeholder=\\"newsletter.firstname.placeholder\\" type=\\"text\\" name=\\"FNAME\\" class=\\"form-control\\"></v-text-field-stub>
        </div>
        <div class=\\"mc-field-group form-group pb-4\\"><label for=\\"mce-LNAME\\">newsletter.lastname.label</label>
          <v-text-field-stub errorcount=\\"1\\" errormessages=\\"\\" messages=\\"\\" rules=\\"\\" successmessages=\\"\\" backgroundcolor=\\"\\" id=\\"mce-LNAME\\" loaderheight=\\"2\\" clearicon=\\"$clear\\" placeholder=\\"newsletter.lastname.placeholder\\" type=\\"text\\" name=\\"LNAME\\" class=\\"form-control\\"></v-text-field-stub>
        </div>
        <div id=\\"mce-responses\\" class=\\"clear\\">
          <div id=\\"mce-error-response\\" class=\\"response\\" style=\\"display: none;\\"></div>
          <div id=\\"mce-success-response\\" class=\\"response\\" style=\\"display: none;\\"></div>
        </div>
        <div aria-hidden=\\"true\\" style=\\"position: absolute; left: -5000px;\\"><input type=\\"text\\" name=\\"b_7f69c361c7f5d97725d2e339b_c821016b08\\" tabindex=\\"-1\\" value=\\"\\"></div>
        <div class=\\"clear\\">
          <v-btn-stub color=\\"primary\\" tag=\\"button\\" activeclass=\\"\\" type=\\"submit\\" id=\\"mc-embedded-subscribe\\" name=\\"subscribe\\" class=\\"button\\">newsletter.submit</v-btn-stub>
        </div>
      </div>
    </form>
  </div>
</div>"
`;
